File: app/src/main/java/com/github/alexeygorovoy/moxy_template/ui/common/moxy/BaseMvpView.java
package com.github.alexeygorovoy.moxy_template.ui.common.moxy;

import android.support.annotation.NonNull;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

public interface BaseMvpView extends MvpView {
    @StateStrategyType(SkipStrategy.class)
    void showInfoToast(@NonNull CharSequence message);

    @StateStrategyType(SkipStrategy.class)
    void showErrorToast(@NonNull CharSequence message);

    @StateStrategyType(SkipStrategy.class)
    void showSuccessToast(@NonNull CharSequence message);
}

File: app/src/main/java/com/github/alexeygorovoy/moxy_template/ui/common/moxy/BaseMvpView.kt
package com.github.alexeygorovoy.moxy_template.ui.common.moxy

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

interface BaseMvpView : MvpView {
    @StateStrategyType(SkipStrategy::class)
    fun showInfoToast(message: CharSequence)

    @StateStrategyType(SkipStrategy::class)
    fun showErrorToast(message: CharSequence)

    @StateStrategyType(SkipStrategy::class)
    fun showSuccessToast(message: CharSequence)
}
